src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -190,6 +193,8 @@ public function handle(Request $request, $type = HttpKernelInterface::MASTER_REQ
             $this->boot();
         }
 
+        $request->setIpRetriever($this->container->get('ip_retriever'));
```

Comment:
very good point. I see how I can inject the ipRetriever in the request through an event listener, but not "when it is necessary". Do I have any mecanism to do any kind of lazy loading here ?